packages/inter-protocol/src/proposals/startPSM.js

```diff
@@ -321,12 +328,20 @@ export const makeAnchorAsset = async (
       },
     }),
   );
-  // TODO: save adminFacet of this contract, like all others.
+
+  const startUpgradeable = await startUpgradeableP;
+  produceAnchorKits.resolve([]);
   /** @type {{ creatorFacet: ERef<Mint<'nat'>>, publicFacet: ERef<Issuer<'nat'>> }} */
   // @ts-expect-error cast
-  const { creatorFacet: mint, publicFacet: issuerP } = E.get(
-    E(zoe).startInstance(mintHolder, {}, terms, undefined, keyword),
-  );
+  const { creatorFacet: mint, publicFacet: issuerP } = await startUpgradeable({
+    installation: mintHolder,
+    label: keyword,
+    terms,
+    privateArgs: undefined,
+    produceResults: {
+      resolve: kit => Promise.resolve(anchorKits).then(kits => kits.push(kit)),
+    },
```

Comment:
produceResults sounds to me like a callback, not a promise. This is a hacky promise since it has no `reject`. consider simplifying,
```suggestion
    produceResults: kit => Promise.resolve(anchorKits).then(kits => kits.push(kit))}
```
actually, it doesn't necessarily have to produce. that's up to the caller, right?
```suggestion
    withResults: kit => Promise.resolve(anchorKits).then(kits => kits.push(kit))}
```
but… the result is what `startUpgradable` is returning. Why do we need that parameter at all?

```
  const { creatorFacet: mint, publicFacet: issuerP } = await startUpgradeable({
    installation: mintHolder,
    label: keyword,
    terms,
    privateArgs: undefined,
   }).then(kit => Promise.resolve(anchorKits).then(kits => kits.push(kit)));
```

Reply:
This is actually the crux of the PR--we consume this method with a required param `produceResults`, instead of consuming `zoe` directly and calling `startInstance`. That way, any instance that is started must have its results saved. Leaving it to promise chaining makes it optional and adds risk that we lose an admin facet that needs to be upgraded in the future. Related to some of your other comments, we don't necessarily expect we'll need to upgrade all these instances in the future, but we're storing the admin facets this way to be safe.

Reply:
Yeah, there's the helper. I'm saying that giving the callback isn't necessary since the helper is returning a promise. Specifically, this would work,
```js
  const upgradable = startUpgradeable({
    installation: mintHolder,
    label: keyword,
    terms,
    privateArgs: undefined,
  });
  upgradable.then(kit => (await anchorKits).push(kit));
  /** @type {{ creatorFacet: ERef<Mint<'nat'>>, publicFacet: ERef<Issuer<'nat'>> }} */
  // @ts-expect-error cast
  const { creatorFacet: mint, publicFacet: issuerP } = await upgradable;
```
I find the `kit => (await anchorKits).push(kit)` a lot easier to read than `produceResults: kit => Promise.resolve(anchorKits).then(kits => kits.push(kit))}` but YMMV.

This would also work and I think an unequivocal improvement to readability (and not hacking the Promise type):
```js
  const { creatorFacet: mint, publicFacet: issuerP } = await startUpgradeable({
    installation: mintHolder,
    label: keyword,
    terms,
    privateArgs: undefined,
    withResults: kit => Promise.resolve(anchorKits).then(kits => kits.push(kit)),
  });
```
Incidentally, shouldn't `anchorKits` be awaited before this to ensure that it doesn't fail after the contract is made? Then it could be,
```js
  const { creatorFacet: mint, publicFacet: issuerP } = await startUpgradeable({
    installation: mintHolder,
    label: keyword,
    terms,
    privateArgs: undefined,
    withResults: kit => anchorKits.push(kit)),
  });
```
…which reads very clearly to me.

Reply:
> `Promise.all([upgradable, anchorKits]).then(([kit, kits]) => kits.push(kit));`

Again, this seems to defeat the purpose of making this util function. I could be misjudging though though (cc @dckc) 

There's other uses of `produceResults` where we do pass in a promise though. I suppose we could make the function take either a promise or a `withResults` function param as you suggest if necessary for readability, though I'd argue that makes it less clear that the param is intended for resolving a value.

Reply:
```js
  const upgradable = startUpgradeable({
    installation: mintHolder,
    label: keyword,
    terms,
    privateArgs: undefined,
  });
```

How does that enforce that the `adminFacet` is saved?

The case of multiple calls to one `startUpgradeable` call site is messy. Perhaps there should be an optional `key` arg, and if present, `startUpgradeable` would take care of storing in a collection under that key. For example, `psmKit` is keyed by brand. And maybe some way for the caller to signal "just push it on an array; I don't have a key".

Reply:
> startUpgradeable would take care of storing in a collection under that key

Yeah that's what I expected the function was for. I was so sure of it that I didn't confirm it's actually doing that. Let's chat to get on the same page.

Reply:
… chatted and I agree that this design is appropriate given current constraints. It solves the requirements and we can come back later to refactor.